tests/integration/transforms/go/simple/main.go

```diff
@@ -0,0 +1,186 @@
+				return &pulumi.XResourceTransformResult{
+					Props: rta.Props,
+					Opts:  rta.Opts,
+				}
+			},
+		}))
+		if err != nil {
+			return err
+		}
+
+		// Scenario #2 - apply a transform to a Component to transform it's children
+		_, err = NewMyComponent(ctx, "res2", pulumi.XTransforms([]pulumi.XResourceTransform{
+			func(rta *pulumi.XResourceTransformArgs) *pulumi.XResourceTransformResult {
+				fmt.Printf("res2 transform\n")
+				if rta.Type == "testprovider:index:Random" {
+					rta.Opts.AdditionalSecretOutputs = append(rta.Opts.AdditionalSecretOutputs, "result")
+					return &pulumi.XResourceTransformResult{
+						Props: rta.Props,
+						Opts:  rta.Opts,
+					}
+				}
+				return nil
+			},
+		}))
+		if err != nil {
+			return err
+		}
+
+		// Scenario #3 - apply a transform to the Stack to transform all (future) resources in the stack
+		err = ctx.XRegisterStackTransform(func(rta *pulumi.XResourceTransformArgs) *pulumi.XResourceTransformResult {
+			fmt.Printf("stack transform\n")
+			fmt.Printf("%v %v\n", rta.Type, rta.Props)
+			if rta.Type == "testprovider:index:Random" {
+				rta.Props["prefix"] = pulumi.String("stackDefault")
+				rta.Opts.AdditionalSecretOutputs = append(rta.Opts.AdditionalSecretOutputs, "result")
+
+				return &pulumi.XResourceTransformResult{
+					Props: rta.Props,
+					Opts:  rta.Opts,
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = NewRandom(ctx, "res3", &RandomArgs{
+			Length: pulumi.ToSecret(pulumi.Int(5)).(pulumi.IntOutput),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Scenario #4 - transforms are applied in order of decreasing specificity
+		// 1. (not in this example) Child transform
+		// 2. First parent transform
+		// 3. Second parent transform
+		// 4. Stack transform
+		_, err = NewMyComponent(ctx, "res4", pulumi.XTransforms([]pulumi.XResourceTransform{
+			func(rta *pulumi.XResourceTransformArgs) *pulumi.XResourceTransformResult {
+				fmt.Printf("res4 transform\n")
+				if rta.Type == "testprovider:index:Random" {
+					rta.Props["prefix"] = pulumi.String("default1")
+
+					return &pulumi.XResourceTransformResult{
+						Props: rta.Props,
+						Opts:  rta.Opts,
+					}
+				}
+				return nil
+			},
+			func(rta *pulumi.XResourceTransformArgs) *pulumi.XResourceTransformResult {
+				fmt.Printf("res4 transform 2\n")
+				if rta.Type == "testprovider:index:Random" {
+					rta.Props["prefix"] = pulumi.String("default2")
+
+					return &pulumi.XResourceTransformResult{
+						Props: rta.Props,
+						Opts:  rta.Opts,
+					}
+				}
+				return nil
+			},
+		}))
+		if err != nil {
+			return err
+		}
+
+		// Scenario #5 - mutate the properties of a resource
+		_, err = NewRandom(ctx, "res5", &RandomArgs{Length: pulumi.Int(10)}, pulumi.XTransforms([]pulumi.XResourceTransform{
+			func(rta *pulumi.XResourceTransformArgs) *pulumi.XResourceTransformResult {
+				fmt.Printf("res5 transform\n")
```

Comment:
more debugging leftovers?

Reply:
Kinda, the transform test I copied for this had print statements left in as well.